src/averageAges.js

```diff
@@ -55,7 +68,21 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = !onlyWithSon
+    ? people.filter(child => (
+      people.find(mother => mother.name === child.mother))
+    )
+    : people.filter(child => (
+      people.find(mother => mother.name === child.mother) && child.sex === 'm')
+    );
+
+  const averageAgeDiff = children.reduce((sumOfAges, child) => (
+    sumOfAges
+    + child.born
+    - people.find(mother => mother.name === child.mother).born
```

Comment:
Will be easier to read if you create a separate variable for finded mother. But on your choice )